CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/HabitFragments.java

```diff
@@ -0,0 +1,194 @@
+import com.google.android.material.button.MaterialButtonToggleGroup;
+
+import java.util.Date;
+
+public class HabitFragments extends DialogFragment {
+
+    private EditText habitTitle;
+    private EditText startDate;
+    private HabitFragments.OnFragmentInteractionListener listener; //Comforting to have.
+    private TableLayout tableLayout;
+    private Button frequencyButton;
+    private Button dateButton;
+    private CalendarView calendarView;
+
+
+
+    public interface OnFragmentInteractionListener{
+        void onOkPressed(Habit newHabit);
+
+        void onOkPressed(Habit Retrieved_Habit, String name, String reason, Date date , Byte weekdays);
+    }
+
+    @Override
+    public void onAttach(Context context){
+        super.onAttach(context);
+        if (context instanceof OnFragmentInteractionListener) {
+            listener = (OnFragmentInteractionListener) context;
+        } else {
+            throw new RuntimeException((context.toString()) + "must implement OnFragementInteractionListener." );
+        }
+    }
+
+    static HabitFragments newInstance (Habit habit){
+        Bundle args = new Bundle();
+        args.putSerializable("habit", habit);
+        HabitFragments fragment = new HabitFragments();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        //return super.onCreateDialog(savedInstanceState);
+        View view = LayoutInflater.from(getActivity()).inflate(R.layout.activity_habit_fragments, null);
+        habitTitle = view.findViewById(R.id.habitTitleEditText);
+        startDate = view.findViewById(R.id.reasonEditText);
+        tableLayout = view.findViewById(R.id.customSelector);
+        frequencyButton = view.findViewById(R.id.frequencyButton);
+        dateButton = view.findViewById(R.id.dateButton);
+        calendarView = view.findViewById(R.id.calendarView2);
+        calendarView.setVisibility(View.GONE);
+        tableLayout.setVisibility(View.GONE);
+
+
+
+        frequencyButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                tableLayout.setVisibility(tableLayout.getVisibility() == View.GONE ? View.VISIBLE : View.GONE);
+            }
+        });
+        dateButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (calendarView.getVisibility() == View.GONE)
+                    calendarView.setVisibility(View.VISIBLE);
+                else
+                    calendarView.setVisibility(View.GONE);
+                //calendarView.setVisibility(calendarView.getVisibility() == View.GONE ? View.VISIBLE : View.GONE);
+            }
+        });
+
+        Bundle args = getArguments();
+        if (args != null) {
+            Habit Retrieved_Habit = (Habit)args.getSerializable("habit");
```

Comment:
I think we should keep our variable naming convention consistent, others use lower camel case with no underscores